src/main/scala/com/mytask/Crimes.scala

```diff
@@ -0,0 +1,104 @@
+package com.mytask
+/**
+  * @author Victor Pobedrya, pobedrik@gmail.com
+  * @version 1.0
+  */
+
+import java.io.File
+
+import scala.collection.mutable.ArrayBuffer
+
+object Crimes extends  App {
+
+  override def main(args: Array[String]): Unit = {
+    try {
+      if (args(0) != "-d"){
+        throw new ArrayIndexOutOfBoundsException()
+      }
+      val directory_path = args(1).toString.trim
+      val crime_records = ArrayBuffer[Map[String, String]]()
+      val crime_files = get_files_list(directory_path)
+      if (crime_files.isEmpty){
+        throw new NullPointerException("No such directory: "+args(1))
+      }
+      for (file_name <- crime_files) {
+        crime_records.appendAll(load_records(file_name.getAbsolutePath))
+      }
+      val grouped_records = group_records(crime_records)
+      print_records(grouped_records)
+    } catch {
+      case e: ArrayIndexOutOfBoundsException => println("Usage:  java -jar crime.jar -d <absolute_path_to_the_crimes_folder>")
+      case e: NullPointerException => println(e.getMessage)
+    }
+  }
+
+  /**
+    * @return Returns list of files in specified directory
+    */
+  def get_files_list(dir: String):List[File] = {
+
+    val d = new File(dir)
+    if (d.exists && d.isDirectory) {
+      d.listFiles.filter(_.isFile).toList
+    } else {
+      List[File]()
+    }
+  }
+
+  /**
+    * @return Load all records from csv files with non-empty 'Crime ID' in one ArrayBuffer
+    */
+  def load_records(file_name: String): ArrayBuffer[Map[String, String]] = {
+    val valid_crimes = ArrayBuffer[Map[String, String]]()
+    val bufferedSource = io.Source.fromFile(file_name)
+
+    try {
+
+      val file_iterator = bufferedSource.getLines
+      val header_line = file_iterator.next()
+      val header_cols = header_line.split(",").map(_.trim)
+
+      for (line <- file_iterator) {
+        val cols = line.split(",").map(_.trim)
+        val record = header_cols.zip(cols).toMap
+        if (record("Crime ID").nonEmpty) {
+          valid_crimes += record
+        }
+      }
+
+    } finally {
+      bufferedSource.close
+    }
+    valid_crimes
+  }
+
+
+  /**
+    * @return Group records in locations an returns top 5 most criminal locations
+    */
+  def group_records(crime_records: ArrayBuffer[Map[String, String]]): ArrayBuffer[((String, String), ArrayBuffer[Map[String, String]])] = {
+    var grouped_records = crime_records.groupBy(record => (record("Latitude"), record("Longitude")))
+    var most_criminal = ArrayBuffer[((String, String), ArrayBuffer[Map[String, String]])]()
+    for (i <- 1 to 5) {
+      val record = grouped_records.toSeq.maxBy(_._2.size)
+      grouped_records -= record._1
+      most_criminal += record
+    }
+    most_criminal
```

Comment:
why not sort descending and just take first N? 
```scala
grouped_records.toSeq.sortBy(-_._2.size).take(5)
```

Reply:
At the beginning I made sorting, but when program loaded about 340 000 records from all files, programs started work. incredibly slow, that's why I decided to get the biggest 5 records without sorting of all dataset

Reply:
on what computer do you run the program?

Reply:
MacBook Pro =))

Reply:
interesting, my version with sorting takes 3seconds to finish for entire program

Reply:
I found previous version
 ```
def group_records(crime_records: ArrayBuffer[Map[String, String]]): ListMap[(String, String), ArrayBuffer[Map[String, String]]] = {
    val grouped_records = crime_records.groupBy(record => (record("Latitude"), record("Longitude")))
    val ordered_records = ListMap(grouped_records.toSeq.sortBy(_._2.size)(Ordering[Int].reverse):_*)
    ordered_records
  }
```

Reply:
And even, when my program filtered records without Crine ID, and get approximately 190 000 records, it worked really slow. I even did't wait to the end

Reply:
Maybe your call `.take(5)` make sorting lazy? But I am not so familiar wit Scala :)